bin/lib/release.js

```diff
@@ -0,0 +1,105 @@
+//good to have the ability to switch for debugging
+const STABLE_BRANCH       = 'master';
+const DIST_BRANCH         = 'dist';
+const OUTPUT_CSS_FILENAME = 'all.css'
+
+import Q        from  'q';
+import fs       from  'fs';
+import path     from 'path';
+import {spawn}  from 'child-process-promise';
+import colors   from 'colors';
+import glob     from 'glob';
+import mversion from 'mversion';
+
+let pReadFile   = Q.denodeify(fs.readFile);
+let pWriteFile  = Q.denodeify(fs.writeFile);
+let pGlob       = Q.denodeify(glob);
+let pUpdate     = Q.denodeify(mversion.update);
+
+let printProgress = (childProcess)=> {
+  childProcess.stdout.pipe(process.stdout);
+  childProcess.stderr.pipe(process.stderr);
+}
+
+let handleError = (err) => {
+  console.log('-----------------------'.red);
+  console.log(err);
+  console.log('-----------------------'.red);
+  process.exit(1);
+}
+
+
+export default (releaseType) => {
+  return Q.async(function* (){
+
+    try {
+
+      if (!releaseType) {
+        throw new Error('Usage ./bin/release <version-number | major | minor | patch | build>');
+      }
+
+      console.log('- Performing reset'.green);
+      yield spawn('git', ['reset', '--hard']).progress(printProgress);
+
+      console.log(`- Checking out ${STABLE_BRANCH}`.green);
+      yield spawn('git', ['checkout', STABLE_BRANCH]).progress(printProgress);
+
+      console.log('- Running build'.green);
+      yield spawn('npm', ['run', 'build']).progress(printProgress);
+
+      console.log(`- Checking out ${DIST_BRANCH}`.green);
+      yield spawn('git', ['checkout', DIST_BRANCH]).progress(printProgress);
+
+      let ignoreFilePath = path.resolve(__dirname, '../../.gitignore');
+
+      if (!fs.existsSync(ignoreFilePath)) {
+        throw new Error('No .gitignore file');
+      }
+
+      //get all css file data
+      let cssFilePattern = path.resolve(__dirname, '../../styles/**/*.css');
+      let cssFiles = yield pGlob(cssFilePattern);
+      let cssData = yield Q.all(cssFiles.map((filePath)=> pReadFile(filePath, 'utf8')));
```

Comment:
Ah ok.

Have you investigated whether it'd make sense to make the component JS import (i.e. depend on) its CSS? That would make the dependencies more obvious. OTOH I'm not sure how it works with Sass and this build process.

We can review later, it's just something I've been looking at recently in the Grid.